src/lib/y2network/dns_config.rb

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+module Y2Network
+  # DNS configuration (hostname, name servers, etc.).
+  class DNSConfig
```

Comment:
well, when I see all that separation. Maybe it makes more sense to have `Y2Network::DNS` or `Y2DNS` and inside config reader, config and others? Still thinking about reasonable naming and structure to be

1. intuitive
2. easy to read
3. easy to write
4. as short as possible while still above is valid.

Reply:
I also had the same doubts, and liked Y2Routing, Y2DNS but it was argued that they where below y2network namespace, so Y2Network::Routing && Y2Network::DNS looks reasonable to me.